addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,463 @@
+            "totalFEDPayable": None,
+            "totalWithholdingIncomeTaxApplicable": None,
+            "totalCVT": None,
+            "totalDiscount": self._l10n_pk_round_value(total_discount),
+            "reason": invoice.l10n_pk_edi_cancel_reason or "",
+            "reasonRemarks": invoice.l10n_pk_edi_cancel_remarks or "",
+            "items": [
+                self._get_l10n_pk_edi_line_details(index, line, tax_details_per_record.get(line, {}))
+                for index, line in enumerate(lines, start=1)
+            ],
+        }
+        return json_payload
+
+    def _l10n_pk_edi_post_invoice(self, invoice):
+        generate_json = self._l10n_pk_edi_generate_invoice_json(invoice)
+        response = self._l10n_pk_edi_generate(invoice.company_id, generate_json)
+        """
+        Example response with no errors
+        {
+            "$id": "1",
+            "version": "1.0",
+            "statusCode": 200,
+            "errorMessage": "",
+            "result": "900005CLNP5914173522",
+            "timestamp": "2023-12-20T16:59:15",
+            "errors": {
+                "$id": "2",
+                "$values": []
+            }
+        }
+
+        Example response with error messages and details in the `errorMessage`
+        field and the `errors["$values"]` array.
+        {
+            "$id": "1",
+            "version": "1.0",
+            "statusCode": 200,
+            "errorMessage": "Invalid BPOSID Not Found!",
+            "result": "",
+            "timestamp": "2024-08-30T13:35:42",
+            "errors": {
+                "$id": "2",
+                "$values": [
+                    "The SellerNTNCNIC field is required.",
+                    "The SellerProvince field is required."
+                ]
+            }
+        }
+        """
+        if response.get("errorMessage"):
+            return {
+                invoice: {
+                    "success": False,
+                    "error": response["errorMessage"],
+                    "blocking_level": "error",
+                }
+            }
+        json_dump = json.dumps(response)
+        json_name = "%s_einvoice.json" % (invoice.name.replace("/", "_"))
+        attachment = self.env["ir.attachment"].create({
+            "name": json_name,
+            "raw": json_dump.encode(),
+            "res_model": "account.move",
+            "res_id": invoice.id,
+            "mimetype": "application/json",
+        })
+        return {
+            invoice: {
+                "success": True,
+                "attachment": attachment
+            }
+        }
+
+    def _get_l10n_pk_total_tax_values(self, tax_lines, move):
+        sales_tax_total = 0.0
+        non_sales_tax_total = 0.0
+        sale_group_ids = (
+            self._get_l10n_pk_sales_groups(move.company_id)
+            + self._get_l10n_pk_withhold_groups(move.company_id)
+        ).ids
+        for line in tax_lines:
+            if line in sale_group_ids:
```

Comment:
Hummm... `line in sale_group_ids` ?
It's not clear. Is line a number here? Is it the `line_code` from the tax details?
If so, call it `line_code`.